### PR TITLE
docs: add Gateway API documentation

### LINKED ISSUE #4786: add Gateway provisioner documentation
We should add documentation that covers the Gateway provisioner and its functionality (i.e. what's exposed in the ContourDeployment CRD), in a style similar to https://projectcontour.io/docs/v1.22.1/config/fundamentals/ and the rest of the HTTPProxy documentation. Right now there's a guide that covers it but it's not versioned, not exhaustive, and not all that discoverable.

### PR DESCRIPTION
**Pull request by skriss**
Adds a Gateway API document under
"Configuration", and moves some of
the content that was previously in
the guide into it.

Closes #4786.

Feedback on organization, missing content, etc. welcome!

Preview: https://deploy-preview-5433--projectcontour.netlify.app/docs/main/config/gateway-api/

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
deploy preview link: https://deploy-preview-5433--projectcontour.netlify.app/docs/main/config/gateway-api/

File: site/content/docs/main/config/gateway-api.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  port: 443
+```
+
+In dynamic provisioning, the Contour Gateway Provisioner will continuously ensure that the Envoy service and daemonset/deployment are kept in sync with the Gateway Listener configuration.
+In static provisioning, it is up to the platform operator to keep the Envoy resources in sync with the Gateway Listeners.
+
+To get from the Gateway Listener port to the port that Envoy will be configured to listen on, i.e. the container port:
+- add 8000 to the Listener port number
+- if the result is greater than 65535, subtract 65535
+- if the result is less than or equal to 1023, add 1023.
```
Comment (sunjayBhatia) [inline site/content/docs/main/config/gateway-api.md:87]: nit: maybe mention this could still have conflicts, and to check on GW status?

**Review by sunjayBhatia — COMMENTED** [review]
+1 to this structure

I was thinking of adding a "Migrating to Gateway API" doc but that could be a section in this new page, link to it for parts, etc.

File: site/content/docs/main/config/gateway-api.md (hunk 1 above)
Comment (skriss) [reply site/content/docs/main/config/gateway-api.md:87]: https://github.com/projectcontour/contour/pull/5433/commits/d738d078046871a6a9cbb222bd1b34332dc06c0a

**Comment by skriss** [comment]
I marked this ready for review, but we should hold off on merging it until just before the 1.26 release since it updates the non-versioned "Getting Started" page.

File: site/content/docs/main/config/gateway-api.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+Contour implements Gateway API in addition to supporting HTTPProxy and Ingress.
+In particular, Contour aims to support all [core and extended features][2] in Gateway API.
+
+Gateway API has a comprehensive [website and docs][1], so this document focuses primarily on unique aspects of Contour's Gateway API implementation, rather than attempting to reproduce all of the content available on the Gateway API website.
+The reader is suggested to familiarize themselves with the basics of Gateway API before continuing with this doc.
+
+In Contour's Gateway API implementation, a Gateway corresponds 1:1 with a single deployment of Contour + Envoy.
+In other words, each Gateway has its own control plane (Contour) and data plane (Envoy).
+
+See the [Gateway API guide][12] for a step-by-step tutorial on getting Contour installed with Gateway API and using it to route traffic to a service.
```
Comment (sunjayBhatia) [inline site/content/docs/main/config/gateway-api.md:18]: suggestion to separate the concerns of these two documents for the intended reader

```suggestion
The remainder of this document delves into more detail regarding configuration options when using Contour with Gateway API.
If you are looking for a way to get started with Gateway API and Contour, see the [Gateway API guide][12], a step-by-step tutorial on getting Contour installed with Gateway API and using it to route traffic to a service.
```

File: site/content/docs/main/config/gateway-api.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+- if the result is less than or equal to 1023, add 1023.
+
+Note that, in rare corner cases, it's possible to have port conflicts.
+Check the Gateway status to ensure that Listeners have been properly provisioned.
+
+## Routing
+
+Gateway API defines multiple route types.
+Each route type is appropriate for a different type of traffic being proxied to a backend service.
+Contour implements `HTTPRoute`, `TLSRoute`, and `GRPCRoute`.
```
Comment (sunjayBhatia) [inline site/content/docs/main/config/gateway-api.md:96]: tcproute too now!

**Review by sunjayBhatia — COMMENTED** [review]
just a couple comments otherwise lgtm

File: site/content/docs/main/guides/gateway-api.md [hunk 4]
```diff
@@ -184,7 +149,7 @@ The above creates:
 
 See the next section ([Testing the Gateway API](#testing-the-gateway-api)) for how to deploy an application and route traffic to it using Gateway API!
```
Comment (sunjayBhatia) [inline site/content/docs/main/guides/gateway-api.md:150]: looks like this needs an update to #test-routing

**Review by sunjayBhatia — APPROVED** [review]

**Comment by skriss** [comment]
Merging this one since we're about ready to release

**Pull request merged**